system/modules/isotope/library/Isotope/Helper/Scope.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * Isotope eCommerce for Contao Open Source CMS
+ *
+ * Copyright (C) 2009 - 2019 terminal42 gmbh & Isotope eCommerce Workgroup
+ *
+ * @link       https://isotopeecommerce.org
+ * @license    https://opensource.org/licenses/lgpl-3.0.html
+ */
+
+namespace Isotope\Helper;
+
+use Contao\System;
+use Symfony\Component\HttpFoundation\Request;
+/**
+ * Class Scope
+ *
+ * Provide methods to check the current scope.
+ *
+ */
+class Scope
+{
+    /**
+     * Check if the current scope is backend.
+     *
+     * @return bool
+     */
+    public static function isBackend(): bool
+    {
+        $container = System::getContainer();
+        $scopeMatcher = $container->get('contao.routing.scope_matcher');
+        $requestStack = $container->get('request_stack');
+
+
+        return $scopeMatcher->isBackendRequest($requestStack->getCurrentRequest() ?? Request::create(''));
```

Comment:
if no request exists, we should not generate one but return `false` (since it cannot be a backend request on command line for example).

Reply:
Thanks, done